src/infra/dex/paraswap/mod.rs

```diff
@@ -43,11 +43,21 @@ pub struct Config {
 }
 
 impl ParaSwap {
-    pub fn new(config: Config) -> Self {
-        Self {
-            client: super::Client::new(Default::default(), config.block_stream.clone()),
+    pub fn new(config: Config) -> anyhow::Result<Self> {
+        let mut key = reqwest::header::HeaderValue::from_str(&config.api_key)?;
+        key.set_sensitive(true);
+
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert("x-api-key", key);
```

Comment:
before it was upper caps, does this make a difference?

Reply:
Even if it doesn't it's more common to have UPPER-KEBAP-CASE for headers anyway, no?

Reply:
Looks like it does not, since the lowercase is used in other solvers with the same nginx config.

Reply:
The [standard](https://stackoverflow.com/a/5259004) says it doesn't matter.